crates/wasmparser/src/define_types.rs

```diff
@@ -0,0 +1,746 @@
+// Define the core Wasm type hierarchy with the given index type.
+//
+// The index type must satisfy the following constraints:
+//
+// * It must implement `Display`
+//
+// * It must implement `Into<u32>` and `From<u32>`
+//
+// * `$index_type::from(u32::from(index))` must be the identity function.
+//
+// * `u32::from($index_type::from(x))` must also be the identity function.
+//
+// * Its `u32` representation must fit within 20 bits, that is
+//
+//       index.into() <= (1 << 20) - 1
+//
+//   must hold true for all indices.
+macro_rules! define_core_wasm_types {
+    ($index_type:ty) => {
+        /// Represents a recursive type group in a WebAssembly module.
+        #[derive(Debug, Clone)]
+        pub struct RecGroup {
+            pub(crate) types: smallvec::SmallVec<[SubType; 1]>,
+
+            /// Whether or not this rec group was explicitly encoded in the
+            /// binary or was implicitly created for a type that was not
+            /// contained in a `(rec ...)` in the types section.
+            pub(crate) explicit_rec_group: bool,
```

Comment:
Given that this is the only use of `smallvec::` and we still need the explicit `bool` here anyway, could `SmallVec` be dropped as a dep?

Reply:
I find that working with smallvec is much easier and we already have it as a dep in this workspace and in wasmtime. How strongly do you feel here?

Reply:
If there was more to use than just this one location I think I'd agree, but if this is the only location to use `smallvec` I don't think it's worth it personally. Are you planning on adding more usages or is this thie only one?

Reply:
I'll just switch it back to the old `enum`.